File: con-modulos/DoubleDispatch/view/GameView.js
import { console } from '../utils/console.js';
import { BoardView } from './BoardView.js';
import { TurnView } from './TurnView.js';

class GameView {

  #game;
  #boardView;
  #turnView;

  constructor(game) {
    this.#game = game;
    this.#boardView = new BoardView(this.#game.getBoard());
    this.#turnView = new TurnView(this.#game.getTurn());
  }

  play() {
    console.writeln(`----- CONNECT4 -----`);
    this.#turnView.configure();
    let gameFinished;
    do {
      this.#boardView.writeln();
      this.#turnView.play();
      gameFinished = this.#game.isFinished();
      if (!gameFinished) {
        this.#game.changeTurn();
      }
    } while (!gameFinished);
    this.#writeResult();
  }

  #writeResult() {
    this.#boardView.writeln();
    console.writeln(this.#game.isWinner() ? `The winner is the player ${this.#game.getCurrentPlayer().getColor()}` : `Tied Game`);
  }
}

export { GameView };

File: designPatterns/framework/src/main/es/connect4/views/GameView.java
package main.es.connect4.views;

import main.es.connect4.models.Game;
import main.es.connect4.views.menu.LanguageMenu;
import main.es.utils.views.YesNoDialog;

public class GameView extends main.es.utils.framework.GameView<Game> {

    private BoardView boardView;
    private TurnView turnView;

    public GameView(Game game) {
        super(game);
        new LanguageMenu("SELECT LANGUAGE:").interact();
        this.boardView = new BoardView(game.getBoard());
        this.turnView = new TurnView(game.getTurn());
    }

    public void start() {
        this.turnView.configTurn();
        MessageManager.getInstance().writeln("GAME_TITLE");
        this.boardView.writeln();
    }

    public void play() {
        do {
            this.turnView.play();
            this.boardView.writeln();
        } while (!this.boardView.isGameFinished());
        this.turnView.writeResult();
    }

    public boolean resume() {
        YesNoDialog yesNoDialog = new YesNoDialog();
        yesNoDialog.read(MessageManager.getInstance().getMessage("RESUME"));
        if (yesNoDialog.isAffirmative()) {
            this.game.reset();
        }
        return yesNoDialog.isAffirmative();
    }

}
